source/common/common/scalar_to_byte_vector.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <inttypes.h>
+
+#include <vector>
+
+namespace Envoy {
+template <typename T> void pushScalarToByteVector(T val, std::vector<uint8_t>& bytes) {
+  uint8_t* byte_ptr = reinterpret_cast<uint8_t*>(&val);
+  for (unsigned int byte_index = 0; byte_index < sizeof val; byte_index++) {
+    bytes.push_back(*byte_ptr);
```

Comment:
Nit: collapse two lines to `bytes.push_back(*byte_ptr++);`

Reply:
Done in https://github.com/envoyproxy/envoy/pull/4973/commits/b31e6f66f1e9771c7525621f5e4923fa688b9a85